app/web/features/auth/logins/LoginsPage.tsx

```diff
@@ -0,0 +1,112 @@
+import { CircularProgress, Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import { activeLoginsKey } from "features/queryKeys";
+import { RpcError } from "grpc-web";
+import { AUTH, GLOBAL } from "i18n/namespaces";
+import { useTranslation } from "next-i18next";
+import { ListActiveSessionsRes } from "proto/account_pb";
+import { useInfiniteQuery, useMutation } from "react-query";
+import { service } from "service";
+import { timestamp2Date } from "utils/date";
+import makeStyles from "utils/makeStyles";
+
+import LoginCard from "./LoginCard";
+
+const useStyles = makeStyles((theme) => ({
+  loginsContainer: {
+    display: "flex",
+    flexDirection: "column",
+    padding: theme.spacing(4),
+    margin: "0 auto",
+    width: "100%",
+    [theme.breakpoints.up("md")]: {
+      width: "50%",
+    },
+  },
+  card: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+  button: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+export default function LoginsPage() {
+  const classes = useStyles();
+  const { t } = useTranslation([GLOBAL, AUTH]);
+
+  const {
+    isLoading,
+    error,
+    data,
+    hasNextPage,
+    fetchNextPage,
+    isFetchingNextPage,
+    refetch,
+  } = useInfiniteQuery<ListActiveSessionsRes.AsObject, RpcError>(
+    activeLoginsKey,
+    ({ pageParam }) => service.account.listActiveSessions(pageParam),
+    {
+      getNextPageParam: (lastPage) =>
+        lastPage.nextPageToken ? lastPage.nextPageToken : undefined,
+    }
+  );
+
+  const sessions = data?.pages.flatMap((page) => page.activeSessionsList);
+
+  const {
+    error: logoutAllError,
+    isLoading: logoutAllIsLoading,
+    mutate: logoutAll,
+  } = useMutation<void, RpcError>(async () => {
+    await service.account.logOutOtherSessions(true);
+    refetch();
```

Comment:
Instead of using `refetch` to force a refetch and passing it in the props, better put the value that should trigger the refetch in the `useInfiniteQuery1 dependencies array on line 50 above

It looks like like this:

I think probably it's not fetching as you have the syntax wrong on line 50 above. It should be `queryKey: activeLoginsKey`.

Reply:
Just following up on this as I think you tagged me again @aapeliv ? This comment was the big one here, to add the queryKey instead of refetch.

Reply:
I've changed it to not use refetch, but I'm invalidating the queries wherever that changes. I'm not sure if I misunderstood what you meant, if there was another way!